src/site/content/en/blog/app-history-api/index.md

```diff
@@ -247,7 +258,7 @@ appHistory.updateCurrent({ state });
 
 The "current entry" is not all, though.
 The API also provides a way to access the entire list of entries that a user has navigated through while using your site via its `appHistory.entries()` call, which returns a snapshot array of entries.
-This could be used to, e.g., show different UI based on how the user navigated to a certain page, or just to look back at the previous URLs or their state.
+This could be used to, e.g., show a different UI based on how the user navigated to a certain page, or just to look back at the previous URLs or their states.
 This is impossible with the current History API.
 
 Entries themselves support events that allow operations to be taken when the entry finishes loading, is navigated to or from, or if it's disposed of from the user's history (e.g., an entry forward of the current position can be disposed because the user clicked a link and cleared the existing stack).
```

Comment:
```suggestion
Entries themselves support events that allow operations to be taken when the entry finishes loading, is navigated to or from, or if it's disposed of from the user's history. (E.g., an entry forward of the current position can be disposed because the user clicked a link and cleared the existing stack.)
```